### PR TITLE
Fix the issue that tiflash listen on IPv4 addr when `preferIPv6` is enabled

### PR DESCRIPTION
**Pull request by KanShiori**
### Code changes

- [x] Has Go code change
- [ ] Has CI related scripts change

### Tests

- [ ] Unit test 
- [ ] E2E test 
- [ ] Manual test 
- [ ] No code 

### Side effects

- [ ] Breaking backward compatibility 
- [ ] Other side effects: 

### Related changes

- [ ] Need to cherry-pick to the release branch 
- [ ] Need to update the documentation 

### Release Notes

Please refer to [Release Notes Language Style Guide](https://github.com/pingcap/tidb-operator/blob/master/docs/release-note-guide.md) before writing the release note.

```release-note
Fix the issue that tiflash listen on IPv4 addr when `preferIPv6` is enabled
```

### CODE REVIEW

File: pkg/manager/member/tiflash_util.go [hunk 1]
```diff
@@ -199,6 +201,12 @@ func getTiFlashConfigV2(tc *v1alpha1.TidbCluster) *v1alpha1.TiFlashConfigWraper
 			pdAddr = fmt.Sprintf("%s.%s.svc%s:2379", controller.PDMemberName(ref.Name), ref.Namespace, controller.FormatClusterDomain(ref.ClusterDomain)) // use pd of reference cluster
 		}
 		common.SetIfNil("raft.pd_addr", pdAddr)
+
+		if listenHost == listenHostForIPv6 {
+			// In IPv4/IPv6 dual-stack, the default listen host is 0.0.0.0, so we need to set it to [::]
+			common.SetIfNil("listen_host", listenHost)
```
Comment (ywqzzy) [inline pkg/manager/member/tiflash_util.go:207]:
```suggestion
			common.SetIfNil("listen_host", "::")
```

Comment (KanShiori) [reply pkg/manager/member/tiflash_util.go]: Done

**Review by csuzhangxc — APPROVED** [review]

**Review by ywqzzy — APPROVED** [review]

**Comment by KanShiori** [comment]
/testpull-e2e-kind-serial pull-e2e-kind-tikv-scale-simultaneously

**Pull request merged**